### PR TITLE
Feature/teeolu/new source code

### PR DESCRIPTION
**Pull request by teeolu**
#### Type of pull request

Specify the type of pull request.

---

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring

#### The issue

Describe (in details) the purpose of the pull request

This PR addresses the following issues:
=> Inability to create a new source code - users can't easily create a new project as they please, they will have to edit the URL to the root URL to achieve that.
=> Inability to fork a source code - presently, a user can't edit another user project, but should be able to fork another user's project to make a replica and editable version of it. This PR enables a user to be able to do it

---

#### The Solution

Describe (in details) your implementation (how you solved the issue)

To address the issue above, neccessary UI implementations like adding add and fork Icons to the UI, and necessary functionalities were integrated.

---

#### Screenshots (if there are changes of UI)

If any changes affect UI or an image will add more context to the solution, please attach a before/after screenshots below.

Screen Shot 2020-02-23 at 7 25 40 AM

---

#### Task Ticket(s)

Reference link to the issue if any

---

### Developer (PR owner) Checklist

- [x] Tested changes.
- [x] Double-checked target branch

### Reviewer Checklist

- [ ] I understand _why_ and _how_.
- [ ] I have read the code review guide at https://www.notion.so/f9labs/Code-Review-Guide-2de31a969bbb4207992a9cb07d6273e3

### CODE REVIEW

File: src/components/MonacoEditor/SourceCodeHeading.tsx [hunk 1]
```diff
@@ -183,51 +218,81 @@ const SourceCodeHeading: React.FC<IProps> = ({
...
+        {renderTitleMenuOptions()}
+      </div>
+      <div className={`${classes.createSourcecode} ${commonCss.flexRow}`}>
+        {!!ownerId && !isOwner && (
+          <Tooltip title="Fork project" leaveDelay={100} placement="bottom" enterDelay={100}>
+            <IconButton
+              onClick={forkSourcecode}
+              color="secondary"
+              classes={{ root: classes.createSourcecodeButton }}>
+              <DeviceHubIcon className={classes.createSourcecodeIcon} />
```
Comment (gbenga504) [inline src/components/MonacoEditor/SourceCodeHeading.tsx]: I think it's better to call this ForkIcon. Sounds more intuitive.

File: src/components/MonacoEditor/SourceCodeHeading.tsx [hunk 2]
```diff
@@ -183,51 +218,81 @@ const SourceCodeHeading: React.FC<IProps> = ({
...
+            </IconButton>
+          </Tooltip>
+        )}
+        <Tooltip title="Create new project" leaveDelay={100} placement="bottom" enterDelay={100}>
+          <IconButton
+            onClick={createNewSourcecode}
+            color="secondary"
+            classes={{ root: classes.createSourcecodeButton }}>
+            <NoteAddIcon className={classes.createSourcecodeIcon} />
```
Comment (gbenga504) [inline src/components/MonacoEditor/SourceCodeHeading.tsx]: Same as above., It's better to call this AddNewSourceCodeIcon.

**Review by gbenga504 — CHANGES_REQUESTED** [review]
Looks good, thanks. Just some renaming that's all.

File: src/components/MonacoEditor/SourceCodeHeading.tsx (hunk 1 above)
Comment (teeolu) [reply src/components/MonacoEditor/SourceCodeHeading.tsx]: cool then, i will effect the change

**Review by gbenga504 — APPROVED** [review]
LGTM!

**Pull request merged**